roles/rsyslog/roles/input_roles/basics/defaults/main.yml

```diff
@@ -29,45 +27,99 @@ __rsyslog_basics_rules:
   - '{{ __rsyslog_conf_system_input }}'
   - '{{ __rsyslog_conf_network_input }}'
 
-# Debops example configuration options
-# --------------------------------------------
-
-# __rsyslog_conf_local_modules
+# __rsyslog_conf_local_basics_modules
 #
 # List of ``rsyslogd`` modules that enable logs from the local system to be
 # received and parsed by the ``rsyslogd`` daemon.
 __rsyslog_conf_local_basics_modules:
 
-  - name: 'local-basics-modules'
+  - name: 'basics-modules'
     type: 'modules'
     sections:
 
-      - comment: 'Log messages sent to local UNIX socket'
+      - comment: 'Reads kernel messages (the same are read from journald)'
         options: |-
+          module(load="imklog" permitnonkernelfacility="on")
+        state: '{{ "present"
+                  if ("kernel-message" in rsyslog_capabilities)
+                  else "absent" }}'
+
+      - comment: 'System log messages'
+        options: |-
+          # Log messages sent to local UNIX socket
           module(load="imuxsock"    # provides support for local system logging (e.g. via logger command)
           {% if rsyslog_use_imuxsock %}
                  SysSock.Use="on")  # Turn on message reception via local log socket.
           {% else %}
                  SysSock.Use="off") # Turn off message reception via local log socket.
           {% endif %}
-          # local messages are retrieved through imjournal now.
-          module(load="imjournal"             # provides access to the systemd journal
-                 StateFile="{{ rsyslog_work_dir }}/imjournal.state") # File to store the position in the journal
+          # Read from journald
+          {% set done = "false" %}
+          {% for input in logging_inputs %}
+          {%   if done == "false" and input.type == "basics" %}
+          module(load="imjournal"
+                 StateFile="{{ rsyslog_work_dir }}/imjournal.state"
+                 RateLimit.Burst="{{ input.rsyslog_imjournal_ratelimit_burst | d(20000) }}"
+                 RateLimit.Interval="{{ input.rsyslog_imjournal_ratelimit_interval | d(600) }}"
+                 PersistStateInterval="{{ input.rsyslog_imjournal_persist_state_interval | d(10) }}")
+          {%     set done = "true" %}
+          {%   endif %}
+          {% endfor %}
+          {% set ns = namespace(files_output = false) %}
+          {% set rsyslog_outputs = logging_outputs | d([]) %}
+          {% for output in rsyslog_outputs %}
+          {%   if output.type == "files" %}
+          {%     set ns.files_output = true %}
+          {%   endif %}
+          {% endfor %}
+          {% if not ns.files_output %}
+          {%   set rsyslog_outputs = rsyslog_outputs + [{"name": "files", "type": "files"}] %}
+          {% endif %}
+          {% set indict = {} %}
+          {% set input_list = [] %}
+          {% for input in logging_inputs %}
```

Comment:
is `logging_inputs` guaranteed to be set?  Is it set in `defaults/main.yml` or something like that?

Reply:
Thanks.  I should add it to the defaults file.  (it used to be rsyslog_input_TYPE copied from logging_inputs, where empty list was set if logging_inputs is not given.  But not any more...